pkg/resources/database/resource_db_mysql.go

```diff
@@ -0,0 +1,286 @@
+
+	v3 "github.com/exoscale/egoscale/v3"
+	"github.com/exoscale/terraform-provider-exoscale/pkg/config"
+	providerConfig "github.com/exoscale/terraform-provider-exoscale/pkg/provider/config"
+	"github.com/hashicorp/terraform-plugin-framework-timeouts/resource/timeouts"
+	"github.com/hashicorp/terraform-plugin-framework-validators/stringvalidator"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/planmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringplanmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/schema/validator"
+	"github.com/hashicorp/terraform-plugin-framework/types/basetypes"
+)
+
+type MysqlDatabaseResource struct {
+	DBResource
+}
+
+type MysqlDatabaseResourceModel struct {
+	DBResourceModel
+}
+
+var _ resource.Resource = &MysqlDatabaseResource{}
+var _ resource.ResourceWithImportState = &MysqlDatabaseResource{}
+
+func NewMysqlDatabaseResource() resource.Resource {
+	return &MysqlDatabaseResource{}
+}
+
+func (r *MysqlDatabaseResource) Configure(ctx context.Context, req resource.ConfigureRequest, resp *resource.ConfigureResponse) {
+	if req.ProviderData == nil {
+		return
+	}
+	r.client = req.ProviderData.(*providerConfig.ExoscaleProviderConfig).ClientV3
+}
+
+// ImportState implements resource.ResourceWithImportState.
+func (p *MysqlDatabaseResource) ImportState(ctx context.Context, req resource.ImportStateRequest, resp *resource.ImportStateResponse) {
+	resource.ImportStatePassthroughID(ctx, path.Root("id"), req, resp)
+
+	idParts := strings.Split(req.ID, "@")
+
+	if len(idParts) != 2 || idParts[0] == "" || idParts[1] == "" {
+
+		resp.Diagnostics.AddError(
+			"Unexpected Import Identifier",
+			fmt.Sprintf("Expected import identifier with format: service/database_name@zone. Got: %q", req.ID),
+		)
+
+		return
+	}
+
+	databaseID := idParts[0]
+	zone := idParts[1]
+
+	id := strings.Split(databaseID, "/")
+
+	if len(id) != 2 || id[0] == "" || id[1] == "" {
+		resp.Diagnostics.AddError(
+			"Unexpected Import Identifier",
+			fmt.Sprintf("Expected import identifier with format: service/database_name@zone. Got: %q", req.ID),
+		)
+	}
+
+	serviceName := id[0]
+	databaseName := id[1]
+
+	var data MysqlDatabaseResourceModel
+
+	// Set timeouts (quirk https://github.com/hashicorp/terraform-plugin-framework-timeouts/issues/46)
+	var timeouts timeouts.Value
+	resp.Diagnostics.Append(resp.State.GetAttribute(ctx, path.Root("timeouts"), &timeouts)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+	data.Timeouts = timeouts
+
+	data.Id = types.StringValue(databaseID)
+	data.DatabaseName = types.StringValue(databaseName)
+	data.Service = types.StringValue(serviceName)
+	data.Zone = types.StringValue(zone)
+
+	ReadResourceForImport(ctx, req, resp, &data, p.client)
```

Comment:
I don't think this is needed, as stated in the [docs](https://developer.hashicorp.com/terraform/plugin/framework/resources/import) you only need to set required attributes for `Read` to work.
```
Resources can implement the ImportState method, which must either specify enough Terraform state for the Read method to refresh [resource.Resource](https://pkg.go.dev/github.com/hashicorp/terraform-plugin-framework/resource#Resource) or return an error.
```

Reply:
There is a bit of confusion regarding the naming, ReadResourceForImport is a wrapper for `Read` but is named differently due to different signatures. I'll rename for clarity.

Reply:
Understood! I'll update :)

Reply:
After reading the rest of the code in this case it does not matter much as we only have required attributes so `Read` effectively just checks if resource exists. Still if `optional` attribute gets added latter we would have have to remove `Read` as we don't want to set optional value during `import`. Iit wouldn't hurt to update it now.